docs/reference/technology/connectivity/cellular.md

```diff
@@ -0,0 +1,159 @@
+- `CellularPower` for cellular module power control, such as enabling power save.
+- `CellularInformation` to read the cellular module type and firmware version.
+- `CellularSIM` to enter the PIN code and other SIM management functions.
+- `CellularSMS` to read and write SMS messages.
+
+The CellularDevice class encloses cellular APIs. Therefore, to use any cellular API, you need to get CellularDevice first. You can then use CellularDevice to open and close cellular APIs, as Figure 3 illustrates.
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/api-cellular-device.png)<span>Figure 3. Use CellularDevice to open Cellular APIs</span></span>
+
+When an application has opened a cellular API, you can use it to request API methods. For example:
+
+    CellularNetwork *network  = cellularDevice->get_network();
+    if (network) {
+        printf("Local IP address is %s", network->get_ip_address());
+    }
+
+### UDP and TCP sockets
+
+If you want to use UDP or TCP sockets, you need an IP stack. Mbed OS cellular has an option to use either the LWIP stack, which is part of Mbed OS, or to use the IP stack on the cellular module. Figure 4 illustrates IP stack deployment.
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/api-cellular-ip-stack.png)<span>Figure 4. IP stack can be used in PPP or AT mode</span></span>
+
+#### PPP mode with the LWIP stack on Mbed OS
+
+In PPP mode, the LWIP stack is linked as a part of the Mbed OS application. Sockets are implemented on the LWIP stack, and the modem is used over a Point-to-Point Protocol (PPP) link as a plain data pipe.
+
+You can enable PPP mode in the application configuration file:
+
+    "lwip.ppp-enabled": true
+
+#### AT mode with the IP stack on the modem
+
+In AT mode, the modem's internal IP stack is used over an AT link. Sockets are part of the IP stack, so AT commands are used to control sockets. Socket AT commands are modem specific, and you need to implement them on the Mbed OS side within a `CellularStack` class. You can browse `CellularStack` under the cellular/targets folder to learn how your cellular module supports AT sockets.
+
+The AT mode is enabled when the PPP mode is not enabled:
+
+    "lwip.ppp-enabled": false
+
+#### Should you use PPP or AT mode?
+
+Consider the following points when selecting PPP or AT mode:
+
+- Your cellular module may support only AT or PPP mode.
+- PPP mode supports both UDP and TCP sockets.
+- PPP mode does not allow AT commands after connecting to data mode.
+- PPP mode uses the LWIP stack, which uses memory from your Mbed OS application.
+- AT mode typically supports only UDP sockets.
+- AT mode is potentially better optimized for power consumption.
+
+### Non-IP optimizations
+
+The Mbed OS cellular API has also been planned to **support non-IP communication in the near future**. You need to make two changes when using non-IP in your application:
```

Comment:
Query: Do we want to include this before we actually support non-IP communication?

Reply:
Removed Non-IP socket chapter:
https://github.com/AriParkkila/Handbook/commit/28da2d75b081653974d46c4245abe1a4b025771f